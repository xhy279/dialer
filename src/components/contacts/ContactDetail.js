import React from 'react';
import { connect } from 'react-redux';

import Header from '../header/Header';
import { selectUser } from '../../actions';
import ProfileCard from '../profile/ProfileCard';

class ContactDetail extends React.Component {

  componentWillReceiveProps(nextProps) {
    nextProps.selectUser(nextProps.selectedUser.id);
  }

  renderPage = (user) => {
    return (
      <div>
        <Header config={{name: user.name}} />
        <ProfileCard user={user} />
      </div>
    )
  }
  

  render () {
    const { selectedUser } = this.props;
    return (
      
      <div>
        { selectedUser ? this.renderPage(selectedUser) : null}
      </div>
     
    );
  }
}

const mapStateToProps = (state, ownProps) => {
  return {
    selectedUser: state.usersInfo.users.filter(user => user.id === Number(ownProps.id))[0],
  }
};

const mapDispatchToProps = dispatch => {
  return {
    selectUser: (id) => dispatch(selectUser(id)),
  }
}
export default connect(mapStateToProps, mapDispatchToProps)(ContactDetail);