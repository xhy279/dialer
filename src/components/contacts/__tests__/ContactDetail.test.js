import { shallow } from 'enzyme';
import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import ContactDetail from '../ContactDetail';

const mockStore = configureStore([]);

describe('connected component test', () => {
  it('should render with given state from redux store', () => {
    let store = mockStore({
      users: [],
      callOpen: false,
      modalOpen: false,
    });
    let wrapper = shallow(
      <Provider store={store}>
        <ContactDetail />
      </Provider>
    )
    expect(wrapper).toMatchSnapshot();
  });
})

