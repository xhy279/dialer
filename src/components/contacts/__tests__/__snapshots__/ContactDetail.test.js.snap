// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`connected component test should render with given state from redux store 1`] = `ShallowWrapper {}`;
